### ISSUE TITLE
Option to automatically install nix-darwin

### ISSUE THREAD

**Issue by srid** [DeterminateSystems/nix-installer#350]
This may be outside the scope of this project, but it would be nice if the installer allowed me to also install [nix-darwin](https://github.com/LnL7/nix-darwin) (flakes enabled) in one go.

If it even generated an example `flake.nix` ([with home-manager integration](https://github.com/srid/nixos-flake/blob/8f7241bd4dd15ee430cc49e77bc58bb6dc30ab78/examples/macos/flake.nix#L27-L42) even), the user can get started right way declaratively configuring their packages and services on the mac.

By extension, it could also support pure home-manager configuration (such as on non-NixOS distros).

**Comment by Hoverbear** [comment]
All of those features would be very nice! We have been even been pondering what some of those features would look like. I think we'll focus on making a good Nix installer first, then consider adding these.

**Comment by Hoverbear** [comment]
I don't think the installer should do anything other than install Nix, that's a hard enough job. nix-darwin and home-manager do a good job on their own, we don't need to take on the extra burden.

**Comment by bestlem** [comment]
I would agree that home-manager does just work.

But nix-darwin does have issues for a plain flake install and needs manual intervention - the primary difficulty is /etc/synthetic.conf and that really is plumbing for nix

**Comment by Hoverbear** [comment]
I have followed https://github.com/LnL7/nix-darwin#flakes-experimental several times and not met issues, I do not recall having to edit `/etc/synthetic.conf` during the process. Am I misremembering?

Could you explain the issues you're meeting with a bit more detail?

**Comment by bestlem** [comment]
I have a working flake I install by
nix run `nix-darwin -- switch --flake .#`

This does everything  except setting up /run - and sometimes needing to add `--extra-experimental-features "nix-command flakes"`

**Comment by Hoverbear** [comment]
Is the flake public perchance? If not, could you give me an idea of a few settings I should try? I don't use Mac or Nix-Darwin.

I'll try to reproduce and see if populating `/run` is something we should be doing.

**Comment by srid** [comment]
Here's a simple nix-darwin flake: https://github.com/srid/nixos-flake/blob/master/examples/macos/flake.nix (you can get rid of the home-manager to get just the bare nix-darwin setup)

**Comment by Hoverbear** [comment]
I've tested this a few times now and `home-manager` sets up `/run` successfully all on it's own. It does however hit this problem:

```
ephemeraladmin@mac-epic-turducken nix-darwin % nix run nix-darwin -- switch --flake ~/.config/nix-darwin
building the system configuration...
setting up /run via /etc/synthetic.conf...
error: Unexpected files in /etc, aborting activation
The following files have unrecognized content and would be overwritten:

  /etc/nix/nix.conf

Please check there is nothing critical in these files, rename them by adding .before-nix-darwin to the end, and then try again.
```

This is possibly getting resolved eventually since #620 made it possible for `home-manager` to detect a known output.

I was able to continue:

```
ephemeraladmin@mac-epic-turducken nix-darwin % sudo rm /etc/nix/nix.conf
ephemeraladmin@mac-epic-turducken nix-darwin % nix run nix-darwin --experimental-features "flakes nix-command" -- switch --flake ~/.c
onfig/nix-darwin
building the system configuration...
user defaults...
setting up user launchd services...
setting up /Applications/Nix Apps...
setting up pam...
applying patches...
setting up /etc...
system defaults...
setting up launchd services...
creating service org.nixos.activate-system
reloading service org.nixos.nix-daemon
reloading nix-daemon...
waiting for nix-daemon
waiting for nix-daemon
configuring networking...
ephemeraladmin@mac-epic-turducken nix-darwin %
```

Is it possible this was fixed in `nix-darwin` perhaps?
